src/main/java/io/confluent/idesidecar/restapi/credentials/BasicCredentials.java

```diff
@@ -0,0 +1,144 @@
+package io.confluent.idesidecar.restapi.credentials;
+
+import static io.vertx.core.http.HttpHeaders.AUTHORIZATION;
+
+import io.confluent.idesidecar.restapi.exceptions.Failure.Error;
+import io.confluent.idesidecar.restapi.models.ConnectionSpec.SecurityProtocol;
+import io.confluent.idesidecar.restapi.models.ConnectionSpec.SslIdentificationAlgorithm;
+import io.quarkus.runtime.annotations.RegisterForReflection;
+import io.vertx.core.MultiMap;
+import jakarta.validation.constraints.NotNull;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+
+@Schema(description = "Basic authentication credentials")
+@RegisterForReflection
+public record BasicCredentials(
+    @Schema(
+        description = "The username to use when connecting to the external service.",
+        maxLength = USERNAME_MAX_LEN,
+        minLength = 1
+    )
+    @NotNull
+    String username,
+
+    @Schema(
+        description = "The password to use when connecting to the external service.",
+        maxLength = Password.MAX_LENGTH,
+        minLength = 1
+    )
+    @NotNull
+    Password password
+) implements Credentials {
+
+  private static final int USERNAME_MAX_LEN = 64;
+
+  private static final String PLAIN_LOGIN_MODULE_CLASS =
+      "org.apache.kafka.common.security.plain.PlainLoginModule";
+
+  @Override
+  public Type type() {
+    return Type.BASIC;
+}
+
+  @Override
+  public Optional<Map<String, String>> kafkaClientProperties(
+      KafkaConnectionOptions options
+  ) {
+    var config = new LinkedHashMap<String, String>();
+    var protocol = options.securityProtocol();
+    if (protocol != null && protocol != SecurityProtocol.UNKNOWN) {
+      config.put("security.protocol", protocol.toString());
+    }
+    var algorithm = options.sslIdentificationAlgorithm();
+    if (algorithm != null && algorithm != SslIdentificationAlgorithm.UNKNOWN) {
+      config.put("ssl.endpoint.identification.algorithm", algorithm.toString());
+    }
+    if (protocol != null) {
+      config.put("sasl.mechanism", "PLAIN");
+    }
```

Comment:
We require both `username` and `password` to be `@NotNull`, which makes a lot of sense. As a consequence, we should set `sasl.mechanism=PLAIN` in any case. What about removing the `if (protocol != null)` condition?

This comment will probably become obsolete if you decide to implement my above suggestion to remove the `SecurityProtocol` config from the API.